File: app/src/main/java/com/pirateman/exercisemanager/interval/IntervalActivity.java
package com.pirateman.exercisemanager.interval;

import android.annotation.TargetApi;
import android.app.FragmentManager;
import android.app.FragmentTransaction;
import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.Observer;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageButton;
import android.widget.TextView;

import com.pirateman.exercisemanager.R;
import com.pirateman.exercisemanager.exercise.Exercise;
import com.pirateman.exercisemanager.exercise.ExerciseDatabase;

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.util.List;

@TargetApi(26)
public class IntervalActivity extends AppCompatActivity implements View.OnClickListener {

    private EditText edtReps;
    private EditText edtSets;
    private EditText edtWeight;

    private String currentDate;

    private Exercise exercise;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_interval);
        exercise = (Exercise) getIntent().getSerializableExtra("exercise");
        TextView tvExerciseName = findViewById(R.id.tvExerciseName);
        tvExerciseName.setText(exercise.getName());

        TextView tvExerciseDate = findViewById(R.id.tvExerciseDate);

        currentDate = LocalDateTime.now().format(DateTimeFormatter.ofPattern("MM/dd/yyyy"));

        tvExerciseDate.setText(currentDate);

        edtReps = findViewById(R.id.edtReps);

        edtSets = findViewById(R.id.edtSets);

        edtWeight = findViewById(R.id.edtWeight);

        Button btnSubmitInterval = findViewById(R.id.btnIntervalSubmit);
        btnSubmitInterval.setOnClickListener(this);
        ImageButton imgAddInterval = findViewById(R.id.imgAddInterval);
        imgAddInterval.setOnClickListener(this);


        IntervalDao intervalDao = ExerciseDatabase.getINSTANCE(this).intervalDao();
        LiveData<List<Interval>> intervals = intervalDao.getIntervalsByExerciseId(exercise.getId());

        intervals.observe(this, new Observer<List<Interval>>() {
            @Override
            public void onChanged(@Nullable List<Interval> interval) {


            }
        });


        setUpMultiSliderListener();
    }

    public void setUpMultiSliderListener() {
//        MultiSlider slider = findViewById(R.id.msRange);

//        final TextView tvValue = findViewById(R.id.tvValue);
//        tvValue.setText(String.valueOf(slider.getMin()));

//        slider.setOnThumbValueChangeListener(new MultiSlider.OnThumbValueChangeListener() {
//            @Override
//            public void onValueChanged(MultiSlider multiSlider, MultiSlider.Thumb thumb, int thumbIndex, int value) {
//                tvValue.setText(String.valueOf(value));
//            }
//        });
    }

    @Override
    public void onClick(View v) {

        FragmentManager fragmentManager = getFragmentManager();
        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();

        switch (v.getId()) {
            case R.id.btnIntervalSubmit:
                int reps = Integer.valueOf(edtReps.getText().toString());
                int sets = Integer.valueOf(edtSets.getText().toString());
                int weight = Integer.valueOf(edtWeight.getText().toString());

                Interval interval = new Interval();
                interval.setDate(currentDate);
                interval.setExerciseId(exercise.getId());
                interval.setReps(reps);
                interval.setSets(sets);
                interval.setWeight(weight);

                ExerciseDatabase.getINSTANCE(this).exerciseDao().insertInterval(interval);
                break;
            case R.id.imgAddInterval:

                fragmentTransaction.add(R.id.intervalRow, IntervalFragment.newInstance());

                fragmentTransaction.commit();

                break;

        }
    }
}

File: app/src/main/java/com/pirateman/exercisemanager/interval/IntervalActivity.kt
package com.pirateman.exercisemanager.interval

import android.annotation.TargetApi
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity
import com.pirateman.exercisemanager.R
import com.pirateman.exercisemanager.databinding.ActivityIntervalBinding
import com.pirateman.exercisemanager.exercise.Exercise
import org.koin.android.ext.android.inject
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

@TargetApi(26)
class IntervalActivity : AppCompatActivity(), View.OnClickListener {
    private var edtReps: EditText? = null
    private var edtSets: EditText? = null
    private var edtWeight: EditText? = null
    private var currentDate: String? = null
    private var exercise: Exercise? = null

    private val intervalBinding by inject<ActivityIntervalBinding>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_interval)
        exercise = intent.getSerializableExtra("exercise") as Exercise?

        intervalBinding.tvExerciseName.text = exercise!!.name

        currentDate = LocalDateTime.now().format(DateTimeFormatter.ofPattern("MM/dd/yyyy"))
        intervalBinding.tvExerciseDate.text = currentDate

        val btnSubmitInterval = findViewById<Button>(R.id.btnWorkoutSubmit)
        btnSubmitInterval.setOnClickListener(this)
        intervalBinding.imgAddInterval.setOnClickListener(this)
        //val intervalDao = ExerciseDatabase.getINSTANCE(this)?.intervalDao()
        //val intervals = intervalDao?.getIntervalsByExerciseId(exercise!!.id)
        //intervals!!.observe(this, Observer { })
        setUpMultiSliderListener()
    }

    fun setUpMultiSliderListener() {
//        MultiSlider slider = findViewById(R.id.msRange);

//        final TextView tvValue = findViewById(R.id.tvValue);
//        tvValue.setText(String.valueOf(slider.getMin()));

//        slider.setOnThumbValueChangeListener(new MultiSlider.OnThumbValueChangeListener() {
//            @Override
//            public void onValueChanged(MultiSlider multiSlider, MultiSlider.Thumb thumb, int thumbIndex, int value) {
//                tvValue.setText(String.valueOf(value));
//            }
//        });
    }

    override fun onClick(v: View) {
        val fragmentManager = supportFragmentManager
        val fragmentTransaction = fragmentManager.beginTransaction()
        when (v.id) {
            R.id.btnWorkoutSubmit -> {
                val reps = Integer.valueOf(edtReps!!.text.toString())
                val sets = Integer.valueOf(edtSets!!.text.toString())
                val weight = Integer.valueOf(edtWeight!!.text.toString())
                val interval = Interval()
                interval.date = currentDate
                interval.exerciseId = exercise!!.id
                interval.reps = reps
                interval.sets = sets
                interval.weight = weight
                //ExerciseDatabase.getINSTANCE(this)?.exerciseDao()?.insertInterval(interval)
            }
            R.id.imgAddInterval -> {
                //fragmentTransaction.add(R.id.intervalRow, IntervalFragment.newInstance())
                fragmentTransaction.commit()
            }
        }
    }
}